src/rhsmlib/facts/pkg_arches.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import print_function, division, absolute_import
+
+#
+# Copyright (c) 2020 ATIX AG
+#
+# This software is licensed to you under the GNU General Public License,
+# version 2 (GPLv2). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+# along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+#
+# Red Hat trademarks are not licensed under GPLv2. No permission is
+# granted to use or replicate Red Hat trademarks that are incorporated
+# in this software or its documentation.
+#
+import logging
+import platform
+import subprocess
+
+from rhsmlib.facts import collector
+
+log = logging.getLogger(__name__)
+
+
+class SupportedArchesCollector(collector.FactsCollector):
+    """
+    Class used for collecting packages architectures of a host
+    """
+
+    def __init__(self, arch=None, prefix=None, testing=None):
+        super(SupportedArchesCollector, self).__init__(
+            arch=arch,
+            prefix=prefix,
+            testing=testing)
+
+    def get_arches_on_debuntu(self):
+        """
+        Try to return content of all supported packages architectures
+        :return: dictionary containing architectures
+        Otherwise empty dictionary is returned.
+        """
+        arches = []
+
+        try:
+            output = subprocess.check_output("dpkg --print-architecture", shell=True)
+            if output != '':
+                arches.append(output.rstrip())
+        except Exception as e:
+            print(e)
+
+        try:
+            output = subprocess.check_output("dpkg --print-foreign-architectures", shell=True)
+            if output != '':
+                arches.append(output.rstrip())
+        except Exception as e:
+            print(e)
+
+        return {
+            'supported_architectures': ','.join(arches)
+        }
+
+    def get_all(self):
+        """
+        Get all architectures of a debian / ubuntu host
+        :return: dictionary containing architectures
+        """
+        dist_name, dist_vers, _codename = platform.linux_distribution(full_distribution_name=False)
```

Comment:
The method: `platform.linux_distribution()` is marked as deprecated since Python 3.5 and it was removed from Python 3.8. It breaks registration on Fedora 32. Thus it would break registration in future version of RHEL too:

```
2020-07-02 17:03:25,473 [ERROR] subscription_manager.py:626271:MainThread @managercli.py:219 - module 'platform' has no attribute 'linux_distribution'
Traceback (most recent call last):
  File "/home/jhnidek/githup/candlepin/subscription-manager/src/subscription_manager/managercli.py", line 1401, in _do_command
    consumer = service.register(
  File "/home/jhnidek/githup/candlepin/subscription-manager/src/rhsmlib/services/register.py", line 99, in register
    facts_dict = self.facts.get_facts()
  File "/home/jhnidek/githup/candlepin/subscription-manager/src/subscription_manager/facts.py", line 78, in get_facts
    facts = collector.get_all()
  File "/home/jhnidek/githup/candlepin/subscription-manager/src/rhsmlib/facts/all.py", line 39, in get_all
    results.update(fact_collector.get_all())
  File "/home/jhnidek/githup/candlepin/subscription-manager/src/rhsmlib/facts/pkg_arches.py", line 70, in get_all
    dist_name, dist_vers, _codename = platform.linux_distribution(full_distribution_name=False)
AttributeError: module 'platform' has no attribute 'linux_distribution'
```

There is `distro` Python module: https://pypi.org/project/distro/, but it needn't to be available on all systems. You could do something like this:

```python
try:
   from distro import linux_distribution
except ImportError:
    if 'linux_distribution' in platform:
        linux_distribution = platform.linux_distribution
    else:
        linux_distribution = None

if 'linux_distribution' is not None:
    # call linux_distribution() and do the rest things
```

But I would rather recommend to use function from our module: https://github.com/candlepin/subscription-manager/blob/master/src/rhsmlib/facts/hwprobe.py#L146. It is platform independent and it will be there for ever. :smiley:

Reply:
Good catch. Thanks.